web-ui/src/components/reviews/periods/ReviewPeriods.jsx

```diff
@@ -199,90 +259,165 @@ const ReviewPeriods = ({ onPeriodSelected, mode }) => {
   }, [csrf, dispatch, setLoading]);
 
   useEffect(() => {
-      const getSelfReviews = async () => {
-        let reviews = {};
-        Promise.all(periods.map(async (period) => {
-            const res = await findSelfReviewRequestsByPeriodAndTeamMember(period, currentUserId, csrf);
-            const data =
-              res &&
-              res.payload &&
-              res.payload.data &&
-              res.payload.status === 200 &&
-              !res.error
-                ? res.payload.data
-                : null;
-            if (data) {
-              reviews[period.id] = data[0];
-            }
-        })).then(()=> setSelfReviews(reviews));
-      };
-      if (csrf && periods && periods.length > 0 && currentUserId && selfReviews == null) {
-        getSelfReviews();
+    const getSelfReviews = async () => {
+      let reviews = {};
+      Promise.all(
+        periods.map(async period => {
+          const res = await findSelfReviewRequestsByPeriodAndTeamMember(
+            period,
+            currentUserId,
+            csrf
+          );
+          const data =
+            res &&
+            res.payload &&
+            res.payload.data &&
+            res.payload.status === 200 &&
+            !res.error
+              ? res.payload.data
+              : null;
+          if (data) {
+            reviews[period.id] = data[0];
+          }
+        })
+      ).then(() => setSelfReviews(reviews));
+    };
+    if (
+      csrf &&
+      periods &&
+      periods.length > 0 &&
+      currentUserId &&
+      selfReviews == null
+    ) {
+      getSelfReviews();
+    }
+  }, [csrf, periods, currentUserId, selfReviews]);
+
+  const onPeriodClick = useCallback(
+    id => {
+      if (selectReviewPeriod(state, id)?.open) {
+        onPeriodSelected(id);
       }
-    }, [csrf, periods, currentUserId, selfReviews]);
+    },
+    [state, onPeriodSelected]
+  );
 
-  const onPeriodClick = useCallback((id) => {
-    if(selectReviewPeriod(state, id)?.open) {
-      onPeriodSelected(id);
-    }
-  }, [state, onPeriodSelected]);
+  const confirmDelete = useCallback(
+    id => {
+      setToDelete(id);
+      setConfirmOpen(true);
+    },
+    [setToDelete, setConfirmOpen]
+  );
 
-  const confirmDelete = useCallback((id) => {
-    setToDelete(id);
-    setConfirmOpen(true);
-  }, [setToDelete, setConfirmOpen]);
+  const handleReviewTemplateChange = event => {
+    const templateId = event.target.value;
+    setPeriodToAdd({
+      ...periodToAdd,
+      reviewTemplateId: templateId
+    });
+  };
+
+  const handleSelfReviewTemplateChange = event => {
```

Comment:
This is a new function that is called when the user selected a Self-Review Template.